pkg/ipamd/ipamd.go

```diff
@@ -306,20 +307,23 @@ func prometheusRegister() {
 
 // containsInsufficientCIDRsOrSubnetIPs returns whether a CIDR cannot be carved in the subnet or subnet is running out of IP addresses
 func containsInsufficientCIDRsOrSubnetIPs(err error) bool {
-	var awsErr awserr.Error
+	log.Debugf("containsInsufficientCIDRsOrSubnetIPs encountered %v", err)
+	var apiErr smithy.APIError
 	// IP exhaustion can be due to Insufficient Cidr blocks or Insufficient Free Address in a Subnet
 	// In these 2 cases we will back off for 2 minutes before retrying
-	if errors.As(err, &awsErr) {
-		log.Debugf("Insufficient IP Addresses due to: %v\n", awsErr.Code())
-		return awsErr.Code() == INSUFFICIENT_CIDR_BLOCKS || awsErr.Code() == INSUFFICIENT_FREE_IP_SUBNET
+	if errors.As(err, &apiErr) {
```

Comment:
we should test whether those error check is still true with sdkv2.

Reply:
The Error messages themselves are service errors, those remain true. SDK exposes them via interface like `ErrorCode()` instead of `Code()`.  I had verified with `OperationsError`. Relied in Integration tests for the others.